---
title: "<code>chromReads()</code>"
author: "AdriÃ  Mitjavila Ventura"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: true
    toc_float: true
    number_sections: true
    self_contained: yes
    mode: selfcontained
vignette: >
  %\VignetteIndexEntry{chromReads()}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.align = "center", fig.width = 4, fig.height = 4)
devtools::load_all("..")
```

# Load `plotmics`

```{r results=FALSE}
# load plotmics
library(plotmics)
```

# Run `chromReads()`

`chromReads()` draws an horizontal barplot with the number and percentage of reads mapped to each chromosome. 

It takes the path to a BAM file as an input. This BAM file must have a valid index.

```{r include=FALSE}
chromReads(bamfile = "../testdata/sampleX.bam")
```

```{r}
chromReads(bamfile = "../testdata/sampleX.bam")
```
