<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualization of Omics And Sequencing Data In R â€¢ plotmics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Visualization of Omics And Sequencing Data In R">
<meta property="og:description" content="The goal of this package is to provide simple functions to visualize several types of omics and sequencing data.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">plotmics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">7.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/00-install_and_sessioninfo.html">Install plotmics + Session info</a>
    </li>
    <li>
      <a href="articles/01-chromReads.html">chromReads()</a>
    </li>
    <li>
      <a href="articles/02-barDEGs.html">barDEGs()</a>
    </li>
    <li>
      <a href="articles/03-volcanoPlot.html">volcanoPlot()</a>
    </li>
    <li>
      <a href="articles/04-DEcompare.html">DEcompare()</a>
    </li>
    <li>
      <a href="articles/05-barAnno.html">barAnno()</a>
    </li>
    <li>
      <a href="articles/06-upSetPeaks.html">upsetPeaks()</a>
    </li>
    <li>
      <a href="articles/07-ggUpsetPeaks.html">ggUpsetPeaks()</a>
    </li>
    <li>
      <a href="articles/08-ggVennPeaks.html">ggVennPeaks()</a>
    </li>
    <li>
      <a href="articles/09-expressionHeatmap.html">expressionHeatmap()</a>
    </li>
    <li>
      <a href="articles/10-expressionHeatmap2.html">expressionHeatmap2()</a>
    </li>
    <li>
      <a href="articles/11-chromRegions.html">chromRegions()</a>
    </li>
    <li>
      <a href="articles/12-circleRegions.html">circleRegions()</a>
    </li>
    <li>
      <a href="articles/13-ggVennBed.html">ggVennBed()</a>
    </li>
    <li>
      <a href="articles/14-expressionCor.html">expressionCor()</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amitjavilaventura/plotmics/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="plotmics-">
<code>plotmics</code> <img src="logo.png" align="right" alt="" width="350"><a class="anchor" aria-label="anchor" href="#plotmics-"></a>
</h1></div>
<p><em>pl<strong>O</strong>t<strong>MICS</strong></em>: Visualization of omics and sequencing data in R.</p>
<div class="section level2">
<h2 id="information">Information<a class="anchor" aria-label="anchor" href="#information"></a>
</h2>
<p><em>pl<strong>O</strong>t<strong>MICS</strong></em> is an R package to visualize omics and sequencing data in R using simple but useful functions:</p>
<ul>
<li>
<code><a href="reference/chromReads.html">chromReads()</a></code>: draws a <code>ggplot2</code>-based horizontal barplot with the number and the percentage of reads mapped to each chromosome.</li>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: draws a <code>ggplot2</code>-based horizontal barplot with the upregulated and downregulated genes coming from DESeq2.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: draws a <code>ggplot2</code>-based volcano plot with the DE data coming out from DESeq2.</li>
<li>
<code><a href="reference/DEcompare.html">DEcompare()</a></code>: draws a <code>ggplot2</code>-based scatter plot comparing the Log2FCs of two different contrasts.</li>
<li>
<code><a href="reference/barAnno.html">barAnno()</a></code>: takes a list of modified outputs from <code><a href="https://rdrr.io/pkg/ChIPseeker/man/annotatePeak.html" class="external-link">ChIPseeker::annotatePeak()</a></code>, computes the proportion of peaks in distal or promoter regions and draws a <code>ggplot2</code>-based barplot with the correseponding proportion for each sample.</li>
<li>
<code><a href="reference/getVennCounts.html">getVennCounts()</a></code>: helper function that calls <code><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/makeVennDiagram.html" class="external-link">ChIPpeakAnno::makeVennDiagram()</a></code> to intersect different sets of peaks and returns the Venn counts and a list of the peaks present in each set of peaks. Note that the intersection of peaks may have a smaller number than expected to prevent the intersection being bigger than one of the sets.</li>
<li>
<code><a href="reference/upsetPeaks.html">upsetPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws an UpSet plot using the Venn counts and the <code>UpSetR</code> package.</li>
<li>
<code><a href="reference/ggUpsetPeaks.html">ggUpsetPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws a <code>ggplot2</code>-based UpSet plot using the Venn counts.</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws a Venn diagram using the package <code>ggvenn</code>.</li>
<li>
<code><a href="reference/plotDendogram.html">plotDendogram()</a></code>: helper function to draw a dendogram for the heatmaps in <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>.</li>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code>: function that takes a data frame of expression data (including <em>Geneid</em>) and plots a heatmap of the selected genes. It can be any expression data, but it is better for expression values, counts, normalized counts in which the frist column is the <em>Geneid</em> and the other columns are each of the samples that are to be plotted.</li>
<li>
<code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: function that takes a list of data frames with the columns <em>Geneid</em> and <em>log2FoldChange</em> (it can be another type data, such as FPKMs, but its easier to maintain the name of <em>log2FoldChange</em> since it is how it output from <code>DESeq2</code>) and plots a heatmap with the selected genes. It is better for <em>log2FoldChange</em> heatmaps in which each element in the list is a data frame that outputs from <code>DESeq2</code>.</li>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: function that takes the chromosome size information and a list of regions (e.g.Â BED files) to draw the positions of that regions in the genome.</li>
<li>
<code><a href="reference/circleRegions.html">circleRegions()</a></code>: function that takes a list of chromosome size information and a list of regions (e.g.Â BED files) to draw the positions of that regions in the genome. Similar to <code><a href="reference/chromRegions.html">chromRegions()</a></code>, but in a circular plot, allowing different assemblies, pairing regions, etc.</li>
<li>
<code><a href="reference/expressionCor.html">expressionCor()</a></code>: function that computes the correlation of expression values between samples.</li>
<li>
<code><a href="reference/compareCounts.html">compareCounts()</a></code>: function that draws a scatter plot of gene expression values between 2 samples.</li>
<li>
<code><a href="reference/ggVennBed.html">ggVennBed()</a></code>: draws a Venn diagram with the intersection of two BED files, obtained with <code><a href="https://rdrr.io/pkg/bedtoolsr/man/bt.intersect.html" class="external-link">bedtoolsr::bt.intersect()</a></code>.</li>
<li>
<code><a href="reference/expressionHeatmapBar.html">expressionHeatmapBar()</a></code>: draws a heatmap with expression values and combines it with a barplots of sense/antisense expression values.</li>
</ul>
</div>
<div class="section level2">
<h2 id="install-plotmics">Install <code>plotmics</code>
<a class="anchor" aria-label="anchor" href="#install-plotmics"></a>
</h2>
<p>To install <code>plotmics</code> you have to run the following command in R:</p>
<pre><code><span><span class="co"># if not installed, install the devtools package from CRAN </span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span> <span class="op">}</span></span>
<span></span>
<span><span class="co"># install plotmics from this Github repository </span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"amitjavilaventura/plotmics"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="contributors">Contributors<a class="anchor" aria-label="anchor" href="#contributors"></a>
</h2>
<p>This package has been developed by <a href="https://amitjavilaventura.github.io" class="external-link">AdriÃ  Mitjavila Ventura</a>, with some contributions from <a href="https://github.com/dfernandezperez" class="external-link">dfernandezperez</a>.</p>
<p>If you want to contribute to this package, make a post in the issues section in this repository or fork this repository adding your code and do a pull request.</p>
</div>
<div class="section level2">
<h2 id="cite">Cite<a class="anchor" aria-label="anchor" href="#cite"></a>
</h2>
<p>If you use this package, please cite <a href="https://github.com/amitjavilaventura/plotmics" class="external-link">this repository</a> and give it a star.</p>
</div>
<div class="section level2">
<h2 id="versions">Versions<a class="anchor" aria-label="anchor" href="#versions"></a>
</h2>
<p><code>plotmics</code> versions have the structure of <code>1.2.3</code>. The first number (<code>1</code>, <em>major</em>) implies the addition of a function and/or major changes in the packages; the second number (<code>2</code>,<em>minor</em>) implies the addition of new features to a function and possible corrections; the third number (<code>3</code>, <em>micro</em>) implies the correction of minor bugs or addition of minor features.</p>
<p><code>plotmics</code> version history is shown below:</p>
<ul>
<li>
<p><code>v1.0.0</code>:</p>
<ul><li>First version.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v1.1.0</code>:</p>
<ul>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: Rescale output to remove blank space around the Venn diagram.</li>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: Add possibility to scale (<code><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale()</a></code>) data by rows or columns.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v1.1.1</code>:</p>
<ul>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: Add minor formatting options (remove the gene names, change sizes of texts and titles, change the color of the cell border, etc).</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: Add minor changes in order to make it easier to visualize more peaks sets.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v1.1.2</code>:</p>
<ul>
<li>
<code><a href="reference/chromReads.html">chromReads()</a></code>: Change chromosome filtering method.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: Allow dataframes without <code>DEG</code> column as input.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: Change <code>scale = FALSE</code> for <code>scale = "none"</code>.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v1.1.3</code>:</p>
<ul><li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: Fix error in labelling.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v1.1.4</code>:</p>
<ul>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: Add possibility to color the NA values.</li>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: Change title format in.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v2.0.0</code> <em>(2021-09-22)</em>:</p>
<ul><li>Add new function: <code><a href="reference/chromRegions.html">chromRegions()</a></code>.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v2.1.0</code> <em>(2021-09-26)</em>:</p>
<ul>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Allow to take a list of regions as input.</li>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Allow to order the region sets.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v3.0.0</code> <em>(2021-09-28)</em>:</p>
<ul>
<li>Add new function: <code><a href="reference/circleRegions.html">circleRegions()</a></code>.</li>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Allow to color by different parameters.</li>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Allow to add extra info.</li>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Allow to remove or change size of text in the Y axis.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v3.0.1</code> <em>(2021-09-30)</em>:</p>
<ul>
<li>
<code><a href="reference/circleRegions.html">circleRegions()</a></code>: Fix a minor bug about plotting the chromosome labels.</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: Fix a minor bug that caused an intersection with one region more than expected.</li>
<li>
<code><a href="reference/getVennCounts.html">getVennCounts()</a></code>: Remove <code>pkgcond</code> from required packages.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v3.1.0</code> <em>(2021-10-04)</em>:</p>
<ul><li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>and <code><a href="reference/getVennCounts.html">getVennCounts()</a></code>: Allow to consider strand information.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v3.1.1</code> <em>(2021-10-07)</em>:</p>
<ul>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code> and <code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: Change minor features.</li>
<li>
<code><a href="reference/circleRegions.html">circleRegions()</a></code>: Exclude the chromosomes before looking if the chromosome names in regions are also in the chrom.sizes files.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v3.1.2</code> <em>(2021-10-12)</em>:</p>
<ul>
<li>
<code><a href="reference/circleRegions.html">circleRegions()</a></code>: Write the chromosome labels after drawing all the lines and points, so the labels wonâ€™t be under many layers.</li>
<li>
<code><a href="reference/circleRegions.html">circleRegions()</a></code>: Fix error to allow different assemblies.</li>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: Add the possibility to add title and subtitle, and also to count the total number of genes in each contrast.</li>
<li>
<code><a href="reference/barAnno.html">barAnno()</a></code>: Add the possibility to add percentage/counts.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v4.0.0</code> <em>(2021-10-27)</em>:</p>
<ul>
<li>Add new function: <code><a href="reference/expressionCor.html">expressionCor()</a></code>.</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: Add the possibility to annotate the number of true overlaps for each set of peaks.</li>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: Add the possibility to do a <code><a href="https://rdrr.io/r/stats/prop.test.html" class="external-link">prop.test()</a></code> and add the p-value to the plot.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v5.0.0</code> <em>(2021-11-29)</em>:</p>
<ul>
<li>Add new function: <code><a href="reference/compareCounts.html">compareCounts()</a></code>.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: Add the possibility to draw grid lines.</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: Add the possibility to return the lists of overlapping peaks.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v5.1.0</code> <em>(2021-12-03)</em>:</p>
<ul>
<li>
<code><a href="reference/chromRegions.html">chromRegions()</a></code>: Add the possibility to color the cytogenetic bands.</li>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code>: Add the possibility to have more fields in the data frame so it will be possible to use facets and scale.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: Fix plotting of <code>degLabels</code>.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v6.0.0</code> <em>(2022-02-11)</em>:</p>
<ul><li>Add new function: <code><a href="reference/ggVennBed.html">ggVennBed()</a></code>
</li></ul>
<p><br></p>
</li>
<li>
<p><code>v6.0.1</code> <em>(2022-03-09)</em>:</p>
<ul><li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: change how DEG labels are written and colored, from <code>geom_text()</code> to <code>annotate()</code>.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v6.0.2</code> <em>(2022-06-02)</em>:</p>
<ul><li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: add the option to avoid using the columns <em>log2FoldChange</em> and <em>padj</em> to define the DEGs by turning the parameters <code>log2FC</code> and <code>pval</code> to <code>NULL</code>.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v6.0.3</code> <em>(2022-06-15)</em>:</p>
<ul>
<li>
<code><a href="reference/expressionCor.html">expressionCor()</a></code>: add the option to separate the correlation plot into facets with the <code>by_groups</code> parameter.</li>
<li>
<code><a href="reference/expressionCor.html">expressionCor()</a></code>: allow to avoid the usage of a data frame with a <em>Geneid</em> column.</li>
</ul>
<p><br></p>
</li>
<li>
<p><code>v7.0.0</code> <em>(2022-06-29)</em>:</p>
<ul><li>Add new function: <code><a href="reference/expressionHeatmapBar.html">expressionHeatmapBar()</a></code>.</li></ul>
<p><br></p>
</li>
<li>
<p><code>v7.0.1</code> <em>(devel)</em>:</p>
<ul><li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: change arguments of <code>scale_color_manual()</code>, because it was dropping the unused levels in the legend.</li></ul>
<p><br></p>
</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/amitjavilaventura/plotmics/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing plotmics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Adria Mitjavila Ventura <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Adria Mitjavila Ventura.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
