<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualize Omics And Seequencing Data In R • plotmics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Visualize Omics And Seequencing Data In R">
<meta property="og:description" content="The goal of this package is to provide simple functions to visualize several types of omics and sequencing data.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">plotmics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/00-install.html">Install plotmics</a>
    </li>
    <li>
      <a href="articles/01-chromReads.html">chromReads()</a>
    </li>
    <li>
      <a href="articles/02-barDEGs.html">barDEGs()</a>
    </li>
    <li>
      <a href="articles/03-volcanoPlot.html">volcanoPlot()</a>
    </li>
    <li>
      <a href="articles/04-DEcompare.html">DEcompare()</a>
    </li>
    <li>
      <a href="articles/05-barAnno.html">barAnno()</a>
    </li>
    <li>
      <a href="articles/06-upSetPeaks.html">upsetPeaks()</a>
    </li>
    <li>
      <a href="articles/07-ggUpsetPeaks.html">ggUpsetPeaks()</a>
    </li>
    <li>
      <a href="articles/08-ggVennPeaks.html">ggVennPeaks()</a>
    </li>
    <li>
      <a href="articles/09-expressionHeatmap.html">expressionHeatmap()</a>
    </li>
    <li>
      <a href="articles/10-expressionHeatmap2.html">expressionHeatmap2()</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amitjavilaventura/plotmics/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="plotmics-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#plotmics-" class="anchor"></a><code>plotmics</code> <img src="logo.png" align="right" alt="" width="350">
</h1></div>
<p>Visualize omics and sequencing data in R.</p>
<div id="information" class="section level2">
<h2 class="hasAnchor">
<a href="#information" class="anchor"></a>Information</h2>
<p>The goal of this package is to provide simple functions to visualize several omics and sequencing data:</p>
<ul>
<li>
<code><a href="reference/chromReads.html">chromReads()</a></code>: draws a <code>ggplot2</code>-based horizontal barplot with the number and the percentage of reads mapped to each chromosome.</li>
<li>
<code><a href="reference/barDEGs.html">barDEGs()</a></code>: draws a <code>ggplot2</code>-based horizontal barplot with the upregulated and downregulated genes coming from DESeq2.</li>
<li>
<code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>: draws a <code>ggplot2</code>-based volcano plot with the DE data coming out from DESeq2.</li>
<li>
<code><a href="reference/DEcompare.html">DEcompare()</a></code>: draws a <code>ggplot2</code>-based scatter plot comparing the Log2FCs of two different contrasts.</li>
<li>
<code><a href="reference/barAnno.html">barAnno()</a></code>: takes a list of modified outputs from <code><a href="https://rdrr.io/pkg/ChIPseeker/man/annotatePeak.html">ChIPseeker::annotatePeak()</a></code>, computes the proportion of peaks in distal or promoter regions and draws a <code>ggplot2</code>-based barplot with the correseponding proportion for each sample.</li>
<li>
<code><a href="reference/getVennCounts.html">getVennCounts()</a></code>: helper function that calls <code><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/makeVennDiagram.html">ChIPpeakAnno::makeVennDiagram()</a></code> to intersect different sets of peaks and returns the Venn counts and a list of the peaks present in each set of peaks.</li>
<li>
<code><a href="reference/upsetPeaks.html">upsetPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws an UpSet plot using the Venn counts and the <code>UpSetR</code> package.</li>
<li>
<code><a href="reference/ggUpsetPeaks.html">ggUpsetPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws a <code>ggplot2</code>-based UpSet plot using the Venn counts.</li>
<li>
<code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code>: calls <code><a href="reference/getVennCounts.html">getVennCounts()</a></code> and draws a Venn diagram using the package <code>ggvenn</code>.</li>
<li>
<code><a href="reference/plotDendogram.html">plotDendogram()</a></code>: helper function to draw a dendogram for the heatmaps in <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>.</li>
<li>
<code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code>: function that takes a data frame of expression data (including <em>Geneid</em>) and plots a heatmap of the selected genes. It can be any expression data, but it is better for expression values, counts, normalized counts in which the frist column is the <em>Geneid</em> and the other columns are each of the samples that are to be plotted.</li>
<li>
<code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>: function that takes a list of data frames with the columns <em>Geneid</em> and <em>log2FoldChange</em> (it can be another type data, such as FPKMs, but its easier to maintain the name of <em>log2FoldChange</em> since it is how it output from <code>DESeq2</code>) and plots a heatmap with the selected genes. It is better for <em>log2FoldChange</em> heatmaps in which each element in the list is a data frame that outputs from <code>DESeq2</code>.</li>
</ul>
</div>
<div id="install-plotmics" class="section level2">
<h2 class="hasAnchor">
<a href="#install-plotmics" class="anchor"></a>Install <code>plotmics</code>
</h2>
<p>To install <code>plotmics</code> you have to run the following command in R:</p>
<pre><code># if not installed, install the devtools package from CRAN 
if(!require(devtools)) { install.packages("devtools") }

# install plotmics from this Github repository 
devtools::install_github("amitjavilaventura/plotmics")</code></pre>
</div>
<div id="contributors" class="section level2">
<h2 class="hasAnchor">
<a href="#contributors" class="anchor"></a>Contributors</h2>
<p>This package has been developed by <a href="https://amitjavilaventura.github.io">Adrià Mitjavila Ventura</a>, with some contributions from <a href="https://github.com/dfernandezperez">dfernandezperez</a></p>
<p>If you want to contribute to this package, make a post in the issues section in this repository or fork this repository adding your code and do a pull request.</p>
</div>
<div id="cite" class="section level2">
<h2 class="hasAnchor">
<a href="#cite" class="anchor"></a>Cite</h2>
<p>If you use this package, please cite <a href="https://github.com/amitjavilaventura/plotmics">this repository</a> and give it a star.</p>
</div>
<div id="updates" class="section level2">
<h2 class="hasAnchor">
<a href="#updates" class="anchor"></a>Updates</h2>
<ul>
<li>
<code>v1.0.0</code>: first version.</li>
<li>
<code>v1.1.0</code>:
<ul>
<li>Rescale <code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code> output to remove blank space around the Venn diagram.</li>
<li>Add possibility to scale (<code><a href="https://rdrr.io/r/base/scale.html">scale()</a></code>) data by rows or columns in <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>.</li>
</ul>
</li>
<li>
<code>v1.1.1</code>:
<ul>
<li>Add minor formatting options in <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>, such as the possibility to remove the gene names, change sizes of texts and titles, change the color of the border of each cell, etc.</li>
<li>Add minor changes in <code><a href="reference/ggVennPeaks.html">ggVennPeaks()</a></code> in order to make it easier to visualize more peaks sets.</li>
</ul>
</li>
<li>
<code>v1.1.2</code>:
<ul>
<li>Change chromosome filtering method in <code><a href="reference/chromReads.html">chromReads()</a></code>.</li>
<li>Allow dataframes without <code>DEG</code> column as input for <code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>.</li>
<li>Change <code>scale = FALSE</code> for <code>scale = "none"</code> in <code><a href="reference/volcanoPlot.html">volcanoPlot()</a></code>.</li>
</ul>
</li>
<li>
<code>v1.1.3</code>:
<ul>
<li>Fix error in labelling of <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>.</li>
</ul>
</li>
<li>
<code>v1.1.4</code>:
<ul>
<li>Add possibility to color the NA values in <code><a href="reference/expressionHeatmap.html">expressionHeatmap()</a></code> and <code><a href="reference/expressionHeatmap2.html">expressionHeatmap2()</a></code>.</li>
</ul>
</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/amitjavilaventura/plotmics/">https://​github.com/​amitjavilaventura/​plotmics/​</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php">MIT</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Adria Mitjavila Ventura <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Adria Mitjavila Ventura.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
