<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>chromRegions() • plotmics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="chromRegions()">
<meta property="og:description" content="plotmics">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plotmics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00-install.html">Install plotmics</a>
    </li>
    <li>
      <a href="../articles/01-chromReads.html">chromReads()</a>
    </li>
    <li>
      <a href="../articles/02-barDEGs.html">barDEGs()</a>
    </li>
    <li>
      <a href="../articles/03-volcanoPlot.html">volcanoPlot()</a>
    </li>
    <li>
      <a href="../articles/04-DEcompare.html">DEcompare()</a>
    </li>
    <li>
      <a href="../articles/05-barAnno.html">barAnno()</a>
    </li>
    <li>
      <a href="../articles/06-upSetPeaks.html">upsetPeaks()</a>
    </li>
    <li>
      <a href="../articles/07-ggUpsetPeaks.html">ggUpsetPeaks()</a>
    </li>
    <li>
      <a href="../articles/08-ggVennPeaks.html">ggVennPeaks()</a>
    </li>
    <li>
      <a href="../articles/09-expressionHeatmap.html">expressionHeatmap()</a>
    </li>
    <li>
      <a href="../articles/10-expressionHeatmap2.html">expressionHeatmap2()</a>
    </li>
    <li>
      <a href="../articles/11-chromRegions.html">chromRegions()</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amitjavilaventura/plotmics/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="11-chromRegions_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>chromRegions()</h1>
                        <h4 class="author">Adrià Mitjavila Ventura</h4>
            
            <h4 class="date">23 septiembre, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/amitjavilaventura/plotmics/blob/master/vignettes/11-chromRegions.Rmd"><code>vignettes/11-chromRegions.Rmd</code></a></small>
      <div class="hidden name"><code>11-chromRegions.Rmd</code></div>

    </div>

    
    
<div id="load-plotmics" class="section level1">
<h1 class="hasAnchor">
<a href="#load-plotmics" class="anchor"></a>Load <code>plotmics</code>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load plotmics</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://amitjavilaventura.github.io/plotmics/index.html">plotmics</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="run-chromregions" class="section level1">
<h1 class="hasAnchor">
<a href="#run-chromregions" class="anchor"></a>Run <code>chromRegions()</code>
</h1>
<p><code><a href="../reference/chromRegions.html">chromRegions()</a></code> takes a file with the sizes of the chromosomes and draws a <code>ggplot2</code>-based bar plot. Then takes regions in a BED-like format to draw them into each chromosome. The inputs <code>chrom.sizes</code> and <code>regions</code> can be supplied as path to the file where the information is stored or a data frame.</p>
<div id="required-input" class="section level2">
<h2 class="hasAnchor">
<a href="#required-input" class="anchor"></a>Required input</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chrom_sizes</span> <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>
<span class="va">chrom_sizes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span>header <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##   V1        V2
## 1  1 195471971
## 2 10 130694993
## 3 11 122082543
## 4 12 120129022
## 5 13 120421639
## 6 14 124902244</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">regions</span> <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>
<span class="va">regions</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span>header <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##   V1       V2       V3      V4    V5 V6
## 1  1 34813226 34843022 region1 29796  -
## 2  1 57348975 57377520 region2 28545  -
## 3  1 88258855 88277518 region3 18663  -
## 4  1 91403055 91406029 region4  2974  -
## 5  1 91540015 91559583 region5 19568  +
## 6  1 92992344 92997067 region6  4723  -</code></pre>
</div>
<div id="default-run" class="section level2">
<h2 class="hasAnchor">
<a href="#default-run" class="anchor"></a>Default run</h2>
<p>The default run requires only the <code>chrom_sizes</code> and the <code>regions</code> arguments, either as a path to a file or a data frame.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, regions <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-4-1.png" width="384" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chrom_sizes</span> <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span><span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span>header <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">regions</span> <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span>header <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="va">chrom_sizes</span>, regions <span class="op">=</span> <span class="va">regions</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-5-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="second-set-of-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#second-set-of-regions" class="anchor"></a>Second set of regions</h2>
<p>A second set of regions can be included in the plot. In that case, the default coloring by strand will be removed and the coloring will be done based on the region set.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions     <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             regions2    <span class="op">=</span> <span class="st">"../testdata/mm10.regions2.tsv"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-6-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="chromosome-order" class="section level2">
<h2 class="hasAnchor">
<a href="#chromosome-order" class="anchor"></a>Chromosome order</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions     <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             chr_order   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">7</span>,<span class="fl">9</span>,<span class="fl">8</span>,<span class="fl">19</span>,<span class="fl">11</span>,<span class="fl">10</span>,<span class="fl">12</span>,<span class="fl">13</span>,<span class="fl">15</span>,<span class="fl">14</span>,<span class="fl">16</span>,<span class="fl">17</span>,<span class="fl">18</span>,<span class="st">"X"</span>,<span class="st">"Y"</span>,<span class="st">"M"</span>, <span class="st">"MT"</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-7-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="exclude-chromosomes" class="section level2">
<h2 class="hasAnchor">
<a href="#exclude-chromosomes" class="anchor"></a>Exclude chromosomes</h2>
<p>Very often, the genome assemblies of a lot of species have chromosomes/scaffolds with strange names, which are not nice to plot. These can be excluded using the <code>chr_exlude</code> argument with a vector of regular expressions that match the chromosomes to exclude. By default <code>chr_exclude</code> removes the most usuall <em>strange</em> chromosomes, but if you want to remove more chromosomes or don’t want to remove any, you can change the <code>chr_exclude</code> argument.</p>
<p>An example that excludes all the chromosomes that contain a dot in the name:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions     <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             chr_exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"\\."</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-8-1.png" width="384" style="display: block; margin: auto;"></p>
<p>Here you have an example that does not remove any chromosome.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions     <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             chr_exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">" "</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-9-1.png" width="384" style="display: block; margin: auto;"> Note that, since we are converting the chromosome names to an ordered factor with <code>chr_order</code>, the chromosome names that do not appear in <code>chr_order</code> will be groupped and plotted into a <code>NA</code> category.</p>
</div>
<div id="customization" class="section level2">
<h2 class="hasAnchor">
<a href="#customization" class="anchor"></a>Customization</h2>
<div id="flip-axes" class="section level3">
<h3 class="hasAnchor">
<a href="#flip-axes" class="anchor"></a>Flip axes</h3>
<p><code><a href="../reference/chromRegions.html">chromRegions()</a></code> allows flipping the axes by setting the <code>coord_flip</code> argument to TRUE.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes  <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions      <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             coord_flip   <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-10-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="draw-the-points" class="section level3">
<h3 class="hasAnchor">
<a href="#draw-the-points" class="anchor"></a>Draw the points</h3>
<p>By default, <code><a href="../reference/chromRegions.html">chromRegions()</a></code> draws a line/rectangle and a point in the middle of each region. To avoid drawing the points, the argument <code>draw_points</code> can be set to FALSE.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes  <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions      <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             draw_points  <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-11-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="color" class="section level3">
<h3 class="hasAnchor">
<a href="#color" class="anchor"></a>Color</h3>
<p>By default, the regions are colored by strand, if the strand column has “+” and “-” as values. This coloring is controlled with <code>strand_color</code>, which must be a character vector with two colors.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes  <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions      <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             strand_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Black"</span>, <span class="st">"Yellow"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-12-1.png" width="384" style="display: block; margin: auto;"> If you don’t want to color by strand, just turn <code>strand_color</code> to NULL. In this case, the color of the region will be defined by <code>regions_colors</code>, which must be a character vector with two colors. In case that only one region is supplied (<code>regions</code> argument), only the first color will be used, but if a second set of regions is suppplied (<code>regions2</code> argument), both values will be used.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             strand_color  <span class="op">=</span> <span class="cn">NULL</span>,
             regions_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gold3"</span>, <span class="st">"Darkgreen"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-13-1.png" width="384" style="display: block; margin: auto;"></p>
<p>If we supply a second set of regions:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             regions2      <span class="op">=</span> <span class="st">"../testdata/mm10.regions2.tsv"</span>,
             strand_color  <span class="op">=</span> <span class="cn">NULL</span>,
             regions_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gold3"</span>, <span class="st">"Darkgreen"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-14-1.png" width="384" style="display: block; margin: auto;"> Now, imagine that we have regions that do not have a defined strand (e.g. most ChIP-seq peaks). In this case, the <code>strand_color</code> is internally converted to NULL and the <code>regions_colors</code> argument is used to color the regions.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read and format regions file</span>
<span class="va">regions_no_strand</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"../testdata/mm10.regions.tsv"</span>, header <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>V6 <span class="op">=</span> <span class="st">"."</span><span class="op">)</span>

<span class="co"># Draw the plot</span>
<span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="va">regions_no_strand</span>,
             strand_color  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Black"</span>, <span class="st">"Red"</span><span class="op">)</span>,
             regions_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gold3"</span>, <span class="st">"Darkgreen"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-15-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="titles" class="section level3">
<h3 class="hasAnchor">
<a href="#titles" class="anchor"></a>Titles</h3>
<p>Title and subtitle can be supplied through the arguments <code>title</code> and <code>subtitle</code>, respectively. By default, they are set to NULL, but can accept a character of length 1.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             title         <span class="op">=</span> <span class="st">"This is a title"</span>, 
             subtitle      <span class="op">=</span> <span class="st">"This is a subtitle"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-16-1.png" width="384" style="display: block; margin: auto;"></p>
<p>Also, the labels of the axes can be set or removed through the <code>xlab</code> and <code>ylab</code> arguments. By default, the X axis label is set to “Chromosome” and the Y axis is set to “Position (bp)”, but they can be removed setting the corresponding arguments to NULL or changed to any value.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             xlab          <span class="op">=</span> <span class="cn">NULL</span>, 
             ylab          <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-17-1.png" width="384" style="display: block; margin: auto;"> Finally, a caption can be included in the bottom-right corner by setting the <code>caption</code> argument. By default, <code>caption</code> is set to NULL and it can be set to TRUE or any character. If <code>caption</code> is set to a character, whatever is written will be placed in the bottom-right corner. Instead, if it is set to TRUE, what will be written will be the number of regions in the input region sets.</p>
<p>Here there is an example with any character:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             caption       <span class="op">=</span> <span class="st">"This is a caption"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-18-1.png" width="384" style="display: block; margin: auto;"></p>
<p>On the other hand, if <code>caption</code> is set to TRUE, the caption will show the number of regions in the input regions set (<code>regions</code>).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             caption       <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-19-1.png" width="384" style="display: block; margin: auto;"></p>
<p>And, if <code>caption</code> is set to true, when two <code>regions2</code> is also supplied, the number of regions in each region set will be written in the caption.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             regions2      <span class="op">=</span> <span class="st">"../testdata/mm10.regions2.tsv"</span>,
             caption       <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-20-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="legend" class="section level3">
<h3 class="hasAnchor">
<a href="#legend" class="anchor"></a>Legend</h3>
<p>The position of the legend is, by default, the bottom of the plot. This can be changed by changing the <code>legend</code> argument to one of “bottom”, “right”, “top”, “left” or “none” (no legend). The <code>legend</code> argument is passed through <code><a href="https://rpkgs.datanovia.com/ggpubr/reference/theme_pubr.html">ggpubr::theme_pubr()</a></code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             legend        <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-21-1.png" width="384" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/chromRegions.html">chromRegions</a></span><span class="op">(</span>chrom_sizes   <span class="op">=</span> <span class="st">"../testdata/mm10.chrom.sizes"</span>, 
             regions       <span class="op">=</span> <span class="st">"../testdata/mm10.regions.tsv"</span>,
             legend        <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
<p><img src="11-chromRegions_files/figure-html/unnamed-chunk-22-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="further-costumization" class="section level3">
<h3 class="hasAnchor">
<a href="#further-costumization" class="anchor"></a>Further costumization</h3>
<p>Since <code><a href="../reference/chromRegions.html">chromRegions()</a></code> outputs a <code>ggplot2</code>-based bar plot, it can be further customized like any other <code>ggplot2</code>-based plot.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Adria Mitjavila Ventura.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
