<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>barAnno — barAnno • plotmics</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="barAnno — barAnno"><meta property="og:description" content='Function for ChIP-seq and ATAC-seq.
It must be used after the function annotatePeak() from the R package ChIPseeker. @seealso `annotatePeak`
It takes a list of annotation objects that come as output of annotatePeak() and changes the features to "Promoter", "Distal" and "Gene body" (or to "Promoter" and "Distal"). Finally it plots a bargraph with the distribution of all the proportions
As a ggplot2-based function, it allows to add more layers to format the plot.'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plotmics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/00-install_and_sessioninfo.html">Install plotmics + Session info</a>
    </li>
    <li>
      <a href="../articles/01-chromReads.html">chromReads()</a>
    </li>
    <li>
      <a href="../articles/02-barDEGs.html">barDEGs()</a>
    </li>
    <li>
      <a href="../articles/03-volcanoPlot.html">volcanoPlot()</a>
    </li>
    <li>
      <a href="../articles/04-DEcompare.html">DEcompare()</a>
    </li>
    <li>
      <a href="../articles/05-barAnno.html">barAnno()</a>
    </li>
    <li>
      <a href="../articles/06-upSetPeaks.html">upsetPeaks()</a>
    </li>
    <li>
      <a href="../articles/07-ggUpsetPeaks.html">ggUpsetPeaks()</a>
    </li>
    <li>
      <a href="../articles/08-ggVennPeaks.html">ggVennPeaks()</a>
    </li>
    <li>
      <a href="../articles/09-expressionHeatmap.html">expressionHeatmap()</a>
    </li>
    <li>
      <a href="../articles/10-expressionHeatmap2.html">expressionHeatmap2()</a>
    </li>
    <li>
      <a href="../articles/11-chromRegions.html">chromRegions()</a>
    </li>
    <li>
      <a href="../articles/12-circleRegions.html">circleRegions()</a>
    </li>
    <li>
      <a href="../articles/13-ggVennBed.html">ggVennBed()</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/amitjavilaventura/plotmics/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>barAnno</h1>
    <small class="dont-index">Source: <a href="https://github.com/amitjavilaventura/plotmics/blob/HEAD/R/barAnno.R" class="external-link"><code>R/barAnno.R</code></a></small>
    <div class="hidden name"><code>barAnno.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Function for ChIP-seq and ATAC-seq.
It must be used after the function annotatePeak() from the R package ChIPseeker. @seealso <code>`annotatePeak`</code>
It takes a list of annotation objects that come as output of annotatePeak() and changes the features to "Promoter", "Distal" and "Gene body" (or to "Promoter" and "Distal"). Finally it plots a bargraph with the distribution of all the proportions
As a ggplot2-based function, it allows to add more layers to format the plot.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">barAnno</span><span class="op">(</span>
  <span class="va">anno_list</span>,
  anno_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">anno_list</span><span class="op">)</span>,
  names_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">anno_names</span><span class="op">)</span>,
  protein <span class="op">=</span> <span class="cn">NULL</span>,
  protein_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span>,
  main <span class="op">=</span> <span class="cn">NULL</span>,
  subtitle <span class="op">=</span> <span class="cn">NULL</span>,
  ylab <span class="op">=</span> <span class="cn">NULL</span>,
  xlab <span class="op">=</span> <span class="cn">NULL</span>,
  color_palette <span class="op">=</span> <span class="st">"Set2"</span>,
  legend_position <span class="op">=</span> <span class="st">"right"</span>,
  anno_num <span class="op">=</span> <span class="fl">2</span>,
  fill_position <span class="op">=</span> <span class="cn">T</span>,
  xangle <span class="op">=</span> <span class="fl">20</span>,
  width <span class="op">=</span> <span class="fl">0.6</span>,
  counts_label <span class="op">=</span> <span class="cn">F</span>,
  counts_angle <span class="op">=</span> <span class="fl">90</span>,
  counts_size <span class="op">=</span> <span class="fl">3</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>anno_list</dt>
<dd><p>Named list of annotation objects that come from annotatePeak().</p></dd>
<dt>anno_names</dt>
<dd><p>Charachter vector of the same length as 'anno_list'. Names that will be given to each of the objects in anno_list. Not that will be the names plotted in the bargraph</p></dd>
<dt>names_order</dt>
<dd><p>Character vector with the same entries as 'names' with the order wanted to plot the data. Defaults to: unique(names).</p></dd>
<dt>protein</dt>
<dd><p>NULL or haracter vector of the same length as 'anno_list' with the protein chipped in each dataframe of anno_list. If not NULL, these names will be passed through <code>facet_wrap()</code>. Default: NULL.</p></dd>
<dt>protein_order</dt>
<dd><p>Character vector with the same entries as 'protein' with the order wanted to plot the data. Defaults to: unique(protein).</p></dd>
<dt>main</dt>
<dd><p>Character of lenght 1. Title of the plot. Default: NULL.</p></dd>
<dt>subtitle</dt>
<dd><p>Character of lenght 1. Subtitle of the plot. Default: NULL.</p></dd>
<dt>ylab</dt>
<dd><p>Character of lenght 1. Title of the Y axis. Default: NULL.</p></dd>
<dt>xlab</dt>
<dd><p>Character of lenght 1. Title of the X axis. Default: NULL</p></dd>
<dt>legend_position</dt>
<dd><p>Character of lenght 1. Position of the legend. One of c("none", "bottom", "right", "left," "top"). Default: "right"</p></dd>
<dt>anno_num</dt>
<dd><p>Numerical or character of length 1. Number of annotations to plot, either 2 (Promoter/Distal), 3 (Promoter/Gene body/Distal) or 'all' (the annotatePeak() default). Default: 2.</p></dd>
<dt>fill_position</dt>
<dd><p>Logical. If TRUE (default), it the plotted bars will represent proportion of peaks in each feature. If FALSE, the bars will have the height of the total number of peaks with the correspondent feature color.</p></dd>
<dt>xangle</dt>
<dd><p>Numerical of length 1. Angle of the text in the X axis. Default: 20.</p></dd>
<dt>width</dt>
<dd><p>Numerical of length 1. Width of the bar in relative units. Default: 0.6</p></dd>
<dt>counts_label</dt>
<dd><p>Logical of length 1. Whether to write the number and percentage of observations in each annotation. Default: FALSE.</p></dd>
<dt>counts_angle</dt>
<dd><p>Numerical of length 1. Angle of the number and percentage labels. Default: 90</p></dd>
<dt>counts_size</dt>
<dd><p>Numerical of length 1. Size of the number and percentage labels. Default: 3</p></dd>
<dt>palette</dt>
<dd><p>Character of lenght 1. Color palette used to color the bars through the function <code>scale_fill_brewer()</code>. Default: "Set1".</p></dd>
</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/ChIPseeker/man/annotatePeak.html" class="external-link">ChIPseeker::annotatePeak</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>amitjavilaventura</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Adria Mitjavila Ventura.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

